internal/datanode/data_node.go

```diff
@@ -1167,51 +1214,87 @@ func importFlushReqFunc(node *DataNode, req *datapb.ImportTaskRequest, res *root
 			fieldStats = append(fieldStats, &datapb.FieldBinlog{FieldID: k, Binlogs: []*datapb.Binlog{v}})
 		}
 
-		log.Info("now adding segment to the correct DataNode flow graph")
-		// Ask DataCoord to add segment to the corresponding DataNode flow graph.
-		node.dataCoord.AddSegment(context.Background(), &datapb.AddSegmentRequest{
-			Base: &commonpb.MsgBase{
-				SourceID: Params.DataNodeCfg.GetNodeID(),
-			},
-			SegmentId:    segmentID,
-			ChannelName:  segReqs[0].GetChannelName(),
-			CollectionId: req.GetImportTask().GetCollectionId(),
-			PartitionId:  req.GetImportTask().GetPartitionId(),
-			RowNum:       int64(rowNum),
-		})
-
-		binlogReq := &datapb.SaveBinlogPathsRequest{
-			Base: &commonpb.MsgBase{
-				MsgType:   0, //TODO msg type
-				MsgID:     0, //TODO msg id
-				Timestamp: 0, //TODO time stamp
-				SourceID:  Params.DataNodeCfg.GetNodeID(),
-			},
-			SegmentID:           segmentID,
-			CollectionID:        req.GetImportTask().GetCollectionId(),
-			Field2BinlogPaths:   fieldInsert,
-			Field2StatslogPaths: fieldStats,
-			Importing:           true,
+		// Fetch the flow graph on the given v-channel.
+		var ds *dataSyncService
+		// Retry in case the channel hasn't been watched yet.
+		err = retry.Do(context.Background(), func() error {
+			var ok bool
+			ds, ok = node.flowgraphManager.getFlowgraphService(segReqs[0].GetChannelName())
+			if !ok {
+				return errors.New("channel not found")
+			}
+			return nil
+		}, retry.Attempts(getFlowGraphServiceAttempts))
+		if err != nil {
+			log.Error("channel not found in current DataNode",
+				zap.Error(err),
+				zap.String("channel name", segReqs[0].GetChannelName()),
+				zap.Int64("node ID", Params.DataNodeCfg.GetNodeID()))
+			return err
 		}
-
+		// Get the current dml channel position ID, that will be used in segments start positions and end positions.
+		posID, err := ds.getDmlChannelPositionByBroadcast(context.Background(), segReqs[0].GetChannelName(), req.GetBase().GetTimestamp())
```

Comment:
what happens if posID not exist?
because we know this segment has nothing to do with stream data?